sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1519,6 +1524,137 @@ def plot_loading_results(self, subs=None):
+            [0]: cartesian line: 25*SingularityFunction(x, 5, 0)
+            - 25*SingularityFunction(x, 23, 0) + SingularityFunction(x, 30, 1)
+            - 20*SingularityFunction(x, 50, 0) - SingularityFunction(x, 50, 1)
+            + 5 for x over (0.0, 50.0)
+        """
+        if not numpy:
+            raise ImportError("To use this function numpy module is required")
+
+        x = self.variable
+        length = 10 if isinstance(self.length, Symbol) else self.length
+        height = length/10
+
+        rectangles = []
+        rectangles.append({'xy':(0, 0), 'width':length, 'height': height, 'facecolor':"brown"})
+        annotations, markers, load_eq, fill = self._draw_load(pictorial, length)
+        support_markers, support_rectangles = self._draw_supports(length)
+
+        rectangles += support_rectangles
+        markers += support_markers
+
+        sing_plot = plot(height + load_eq, (x, 0, length),
+         xlim=(-height, length + height), ylim=(-length, length), annotations=annotations,
+          markers=markers, rectangles=rectangles, fill=fill, axis=False, show=False)
+
+        return sing_plot
+
+
+    def _draw_load(self, pictorial, length):
+        loads = list(set(self.applied_loads) - set(self._support_as_loads))
+        height = length/10
+        x = self.variable
+
+        annotations = []
+        markers = []
+        scaled_load = 0
+        for load in loads:
+            if load[2] == -1:
+                if load[0].is_positive:
+                    annotations.append({'s':'', 'xy':(load[1], height),  'xytext':(load[1], height*4), 'arrowprops':dict(width= 1.5, headlength=4, headwidth=4, facecolor='black')})
+                else:
+                    annotations.append({'s':'', 'xy':(load[1], 0), 'xytext':(load[1], height - 4*height), 'arrowprops':dict(width= 1.5, headlength=5, headwidth=5, facecolor='black')})
+            elif load[2] == -2:
+                if load[0].is_negative:
+                    markers.append({'args':[[load[1]], [height/2]], 'marker': r'$\circlearrowleft$', 'markersize':15})
+                else:
+                    markers.append({'args':[[load[1]], [height/2]], 'marker': r'$\circlearrowright$', 'markersize':15})
+            elif load[2] >= 0:
+                if pictorial:
+                    value, start, order, end = load
+                    value = 1 if order > 0 else length/2
+                    scaled_load += value*SingularityFunction(x, start, order)
+                    f2 = 1*x**order if order > 0 else length/2*x**order
+                    for i in range(0, order + 1):
+                        scaled_load -= (f2.diff(x, i).subs(x, end - start) *
+                                       SingularityFunction(x, end, i) / factorial(i))
+                    load_args = scaled_load.args
+                else:
+                    if isinstance(load[0], Symbol):
+                        raise ValueError("Magnitude of load: {} should not be a Symbol".format(load))
```

Comment:
This error message seems to be a little misleading. It can be a `Symbol` when `Pictorial` is set to `True`.

Reply:
How about this:
`"Magnitude of load: {} should not be a Symbol when pictorial is False. Try setting pictorial to True".format(load)`

Reply:
Looks good!